internal/core/ports/link_service.go

```diff
@@ -0,0 +1,17 @@
+package ports
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core"
+
+	"github.com/polygonid/sh-id-platform/internal/core/domain"
+)
+
+// LinkService - the interface that defines the available methods
+type LinkService interface {
+	Save(ctx context.Context, ID uuid.UUID, did core.DID, maxIssuance *int, validUntil *time.Time, schemaID uuid.UUID,
```

Comment:
Related to a previous comment regarding ID. The ID should not be a param here. ID should be generated when constructing the new Link entity inside the Save function.  The best point is in the core.Domain constructor or very close to it.